metrics/web/test/features/auth/domain/usercases/receive_authentication_updates_test.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:metrics/features/auth/domain/entities/authentication_exception.dart';
+import 'package:metrics/features/auth/domain/usecases/receive_authentication_updates.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../../test_utils/matcher_util.dart';
+import 'test_utils/user_repository_mock.dart';
+
+void main() {
+  group("ReceiveAuthenticationUpdates", () {
+    test("can't be created with null repository", () {
+      expect(
+        () => ReceiveAuthenticationUpdates(null),
+        MatcherUtil.throwsAssertionError,
+      );
+    });
+
+    test("delegates call to the UserRepository.authenticationStream", () async {
+      final repository = UserRepositoryMock();
```

Comment:
Maybe we can move mock initializing up before tests and reset it using the `reset()` within either `tearDown()` or `setUp()`?